stock_prices/stock_prices.py

```diff
@@ -3,7 +3,18 @@
 import argparse
 
 def find_max_profit(prices):
-  pass
+  sorted_prices = sorted(prices)
+  highest = sorted_prices[-1]
+  count = 0
+  def find_low(l):
+    if prices.index(highest) > prices.index(l):
+      lowest = l
+      return highest - lowest
+    if prices.index(highest) < prices.index(l):
+      nonlocal count
+      count = count + 1
+      return find_low(sorted_prices[count])
+  return find_low(sorted_prices[0])
 
```

Comment:
very interesting approach to the problem, not sure if this would work exactly, you cant sort the problem because there is always an issue of buying and selling in order